torch/_tensor_str.py

```diff
@@ -445,20 +445,29 @@ def _str_intern(inp, *, tensor_contents=None):
         suffixes.append("size=" + str(tuple(self.shape)))
         from torch._subclasses.fake_tensor import FakeTensor
 
-        if not self.is_meta and not isinstance(self, FakeTensor):
+        is_meta = self.is_meta or isinstance(self, FakeTensor)
+        if not is_meta:
```

Comment:
Good question. I guess exposing `nnz` (or not) in meta tensor str representation depends on whether its value is going to be used in the compiler framework.

On the other hand, showing `nnz` in str representation is misleading because there is no `nnz` argument in sparse tensor constructors. In addition, the `nnz` value is represented in the size of `indices` or `values`. So `nnz` would be displayed multiple times if we would enable it here.
I'd leave it as it is for BC.